### ISSUE TITLE
Separate bannedsitelist/bannedurllist files and caches

### ISSUE THREAD

**Issue by GnunuX** [e2guardian/e2guardian#99]
When we set "createlistcachefiles" to "on", the cache for "bannedsitelist" or "bannedurllist" files are created in the same directory as blacklist files.

I've just added a new blacklist, adjust "bannedsitelist" and "bannedurllist" to handle this new blacklist and I have reloaded e2guardian.

As e2guardian runs as "proxy" user, it cannot create ".processed" file (because this directory is not writable for "proxy" user). It raises this kind of error:
"Error creating cache file. Do you have write access to this area: [..]"

I need to "restart" e2guardian or allow the "proxy" user to temporary write in this directory .

We currently have two alternatives:
- I fully restart the e2guardian and then the ".processed" files will be created as "root"
- I directly set the "proxy" as writable under the directory and I reload e2guardian
 
I think these  two options are not an acceptable solution. I think cache file should not write directly in some "/etc" subdirectories but in some "/var/cache" subdirectories.

**Comment by philipianpearce** [comment]
The createlistcachefiles option was created in early versions of dansguardian when processing power was much lower than now, and when the lists had to be unsorted (the 'cache' file is just a sorted version of the file). As this was changed so that dg would accept sorted files and as processor speeds have increased this option is now redundant and in can increase rather than decrease load times. 

I switched this option off many years ago on our systems (both high and low end) due to the bug you have highlighted and have not seen any loss of reload speed as a result. 

I will change the default to 'off' in the next stable release and mark this option as depreciated in the comments. 

It is an option that will be removed in the next major e2g release (v4). 

Regards 

Philip 

----- Original Message -----
